test/units/module_utils/common/parameters/test_handle_aliases.py

```diff
@@ -0,0 +1,95 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2019 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+import pytest
+
+from ansible.module_utils.common.parameters import handle_aliases
+from ansible.module_utils._text import to_native
+
+DEFAULT_LEGAL_INPUTS = [
+    '_ansible_check_mode',
+    '_ansible_debug',
+    '_ansible_diff',
+    '_ansible_keep_remote_files',
+    '_ansible_module_name',
+    '_ansible_no_log',
+    '_ansible_remote_tmp',
+    '_ansible_selinux_special_fs',
+    '_ansible_shell_executable',
+    '_ansible_socket',
+    '_ansible_syslog_facility',
+    '_ansible_tmpdir',
+    '_ansible_verbosity',
+    '_ansible_version',
+]
+
+
+def test_handle_aliases_no_aliases():
+    argument_spec = {
+        'name': {'type': 'str'},
+    }
+
+    params = {
+        'name': 'foo',
+        'path': 'bar'
+    }
+
+    result = (
+        {},
+        DEFAULT_LEGAL_INPUTS + ['name'],
+    )
+
+    assert result == handle_aliases(argument_spec, params)
```

Comment:
It looks like you will need to sort the legal inputs before comparing.

Something like:

```
diff --git a/test/units/module_utils/common/parameters/test_handle_aliases.py b/test/units/module_utils/common/parameters/test_handle_aliases.py
index 0db4c2ee7a..2b1e13cd95 100644
--- a/test/units/module_utils/common/parameters/test_handle_aliases.py
+++ b/test/units/module_utils/common/parameters/test_handle_aliases.py
@@ -59,12 +59,15 @@ def test_handle_aliases_basic():
         'nick': 'foo',
     }
 
-    result = (
+    expected = (
         {'surname': 'name', 'nick': 'name'},
         DEFAULT_LEGAL_INPUTS + ['name', 'surname', 'nick'],
     )
 
-    assert result == handle_aliases(argument_spec, params)
+    expected[1].sort()
+    result = handle_aliases(argument_spec, params)
+    result[1].sort()
+    assert expected == result
 
 
 def test_handle_aliases_value_error():
```

That diff is for the next testa actually.